### ISSUE TITLE
RStudio fails to launch in some enterprise Windows environments

### ISSUE THREAD

**Issue by kevinushey** [rstudio/rstudio#1835]
A number of users have reported RStudio failing to launch in certain Windows enterprise environments. The most common indicator of this issue is this message on startup:

![screen shot 2017-12-11 at 10 46 38 am](https://user-images.githubusercontent.com/1976582/33847800-9fa3029c-de60-11e7-96b0-304123d9cd8c.png)

Most users have reported that the two following actions can resolve the issue:

1. Launch RStudio as an administrator;
2. Delete the client state folder located at `%USERPROFILE%\AppData\Local\RStudio-Desktop`.

Unfortunately, for solution (2), this only alleviates the issue for the next RStudio launch; any subsequent attempts to launch the IDE appear to fail.

My initial hypothesis is that RStudio is failing to read / write some required client state to the `RStudio-Desktop` folder for some reason (perhaps permission related?) and this is inhibiting startup.

**Comment by EKtheSage** [comment]
This is the diagnostic report generated by RStudio

[diagnostics-report.txt](https://github.com/rstudio/rstudio/files/1549016/diagnostics-report.txt)

Originally I would delete the whole RStudio-Desktop flolder to solve the problem, since I didn't want to reset my RStudio settings. So this morning I deleted the files in the following directories and was able to start RStudio: 

- sources/
- monitored/lists/
- log

I haven't figured out which files in those directories were causing RStudio not starting up.

**Comment by kevinushey** [comment]
@EKtheSage, would you mind checking something for me? First, let's get a copy of `RStudio-Desktop` ready that will cause the session to fail to start:

1. Delete the RStudio-Desktop folder,
2. Launch RStudio,
3. Close RStudio.

Then, launch a Windows command prompt (`cmd.exe`), and list the files in that folder:

```
cd %USERPROFILE%\AppData\Local\RStudio_Desktop
tree /F > rstudio-hierarchy.txt
```

Then, copy the contents of that file to a post here, so we can see what's currently in there. In particular, I'm curious whether there's a file called `lock_file` in one of the `sources` subfolders.

If so, this sounds like an issue that might be resolved in the patch release of RStudio, which is currently available as a preview release. Can you try downloading that preview release from:

https://www.rstudio.com/products/rstudio/download/preview/

and let us know if that resolves the issue?

**Comment by EKtheSage** [comment]
I did recall having a lock_file in one of the folders, but I can't see it now. I will report back if I still see the same problem with the preview release. 

In the mean time, this is the output of running the commands in cmd.exe

C:\Users\EXK328\AppData\Local\RStudio-Desktop>type rstudio-hierarchy.txt
Folder PATH listing for volume OSDisk
Volume serial number is DC64-08AB
C:.
│   addin_registry
│   history_database
│   rmd-outputs
│   rstudio-hierarchy.txt
│   saved_source_markers
│
├───client-state
│       environment-grid.persistent
│       workbenchp.persistent
│
├───console06
│       INDEX001
│
├───ctx
├───dictionaries
│   └───custom
├───explorer-cache
├───log
│       rdesktop.log
│       rsession-EXK328.log
│
├───monitored
│   ├───connection_history
│   ├───lists
│   ├───snippets
│   └───user-settings
│           user-settings
│
├───notebooks
│       patch-chunk-names
│
├───pcs
│       files-pane.pper
│       source-pane.pper
│       windowlayoutstate.pper
│       workbench-pane.pper
│
├───presentation
├───profiles-cache
├───projects_settings
│       last-project-path
│
├───sessions
│   └───active
├───sources
│   └───per
│       ├───t
│       └───u
└───viewer-cache

**Comment by kevinushey** [comment]
Does the IDE fail to launch when in this state, still? Or are you able to launch successfully right now?

**Comment by EKtheSage** [comment]
I was able to launch RStudio even without the preview release. So it could be working because the lock_file was not present.

**Comment by hhau** [comment]
G'day,

I am also on Windows 7 enterprise environment, Rstudio Version 1.1.383 / R 3.4.3,  and have been frequently experiencing this bug (although I think it might be a combination of two bugs). 

To reproduce this, I get the Rstudio session to hang by (often unintentionally) *clicking* on an object in the environment pane, which calls `View(object)`. If this object is not a dataframe with data inside it, Rstudio / R will hang and I have to kill the process. 

Upon trying to relaunch Rstudio, it will fail in the manner already detailed in this thread. In order to get rstudio to relaunch I have to remove all the files in ` C:\Users\***\AppData\Local\RStudio-Desktop\sources\s-1E58D14F ` (where ` s-1E58D14F` is the temp folder, i.e. not ./per/ or ./prop/) **except** for the lock_file. Removing the lock_file has no effect on the launch status of Rstudio. I have attached the temp files from within that folder that cause Rstudio to unsuccessfully launch.

Apologies if this issue has already been solved in your dev/daily build, but I thought it might be useful for those whom are having the same issue on 1.1.* builds.

[s-1E58D14F.zip](https://github.com/rstudio/rstudio/files/1605821/s-1E58D14F.zip)

**Comment by kevinushey** [comment]
@hhau Thanks for the extra information. Just to double check, do you see the same issue in the preview release?
 https://www.rstudio.com/products/rstudio/download/preview/

Out of curiosity, are there any special permissions set on these files as they exist on your machine? (I'm curious if it's possible RStudio is getting hung up attempting to read, copy, or move these files because of permission-related reasons)

**Comment by hhau** [comment]
No problem! There are no special permissions on those files. The preview build seems to solve all of the aforementioned issues! Nice. Specifically: 

- Rstudio does not seem to hang when on clicking on an object in the environment pane (calling `View(obj)`)
- 1.1.408 now launches when the above temp files into `/Rstudio-desktop/sources` (which cause 1.1.383 to fail to launch). 

I see the release notes mention fixing issues to do with Windows and networked drives. Even though my R/Rstudio installation is entirely local, I guess the presence of a (number of) networked drive(s) was probably causing some form of issue, which y'all have clearly now identified and fixed. Thanks for pointing me towards the preview version / fixing that!

**Comment by kevinushey** [comment]
It sounds like the issues with launching of RStudio in these environments is fixed with v1.1.419, but there have been some other user reports (e.g. here: https://community.rstudio.com/t/diagnosing-a-non-responsive-rstudio-console/4601/3) of slowness when working with RStudio in these configurations.

I'm going to close this issue as I believe the failure-to-launch issue is resolved; we'll follow up the rest in a new issue.
